### ISSUE TITLE
Windows Share.RequestAsync is brain dead - please make it like Mac, Android and iOS

### ISSUE THREAD

**Issue by cdavidyoung** [dotnet/maui#12822]
### Description

On MacCatalyst a Dropbox transfer is offered. On iPhone and Android there is a rich set of options including cloud services like Google Drive.  Why is Windows limited to email and nearby bluetooth devices?  I need to share files much larger than what will fit in an email.

https://learn.microsoft.com/en-us/dotnet/maui/platform-integration/data/share?view=net-maui-7.0&tabs=ios

### Public API Changes

I don't think the API needs to change.

### Intended Use-Case

My app runs on all 4 of the main Maui platforms.  Windows is currently limited by the Share.RequestAsync call to email for sharing data.  The other platforms allow access to cloud services for saving arbitrarily large files.

**Comment by drasticactions** [comment]
@cdavidyoung I don't believe this is a limitation with MAUI, but rather of the Windows share sheet itself.

https://github.com/dotnet/maui/blob/main/src/Essentials/src/Share/Share.uwp.cs#L72-L98

This is the Essentials implementation of `IDataTransferManagerInterop`. As far as I can tell, this is fine for the usecase given (Sharing text and URLs). The "limited to email and nearby Bluetooth devices" is, most likely, you lack share targets on your system.

https://learn.microsoft.com/en-us/windows/uwp/app-to-app/receive-data

Applications need to declare themselves as share targets in order to receive data. If you don't have one (or the apps you're using have not implemented them) then it won't appear. MAUI can't fix that.

**Comment by jfversluis** [comment]
As indicated, this is not something we can fix but just the way Windows apps work. If you want that to change, provide feedback through Windows.

**Comment by cdavidyoung** [comment]
Yeah, I have both Google Drive and Dropbox installed on my Windows laptop and I don't understand why they don't show up as a share target.  In practical terms, thinking about the end user of a Maui app, they would find it pretty frustrating that the only share option is email.  Is there something we can do as developers to provide more share options?  What is an acceptable workaround?
